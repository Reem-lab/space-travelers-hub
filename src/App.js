import './style/App.css';

function App() {
  return (
    <div className="App">
      <h1>I still work</h1>
    </div>
  );
}

export default App;
