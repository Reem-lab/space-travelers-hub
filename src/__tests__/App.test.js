import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/store';

describe('Test Router behaviour', () => {
  it('When Dragons in navbar is clicked, it renders the Dragons Section', () => {
    // ARRANGE

    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    // ACT
    const dragonsLink = screen.getByText('Dragons');
    fireEvent.click(dragonsLink);

    // ASSERT
    const body = container.querySelector('body');
    expect(body).toMatchSnapshot();
  });
});
