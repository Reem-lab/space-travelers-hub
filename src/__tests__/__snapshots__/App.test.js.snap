// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar"
  >
    <img
      alt="logo-planet"
      className="logo"
      src="planet.png"
    />
    <div
      className="navbar-title"
    >
      <h1
        className="title"
      >
         Space Travellers' Hub 
      </h1>
    </div>
    <ul
      className="menu"
    >
      <li
        className="li-links"
      >
        <a
          activeclassname="active-link"
          aria-current="page"
          className="links active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="li-links"
      >
        <a
          activeclassname="active-link"
          className="links"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="li-links"
      >
        <a
          activeclassname="active-link"
          className="links"
          href="/Dragons"
          onClick={[Function]}
        >
          Dragons
        </a>
      </li>
      <li
        className="li-links"
      >
        <a
          activeclassname="active-link"
          className="links"
          href="/MyProfile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="heading"
    >
      No Rockets Found ❗❕
    </div>
  </div>
</div>
`;
